`test_version_matches_path_scoped_commit_when_no_tag` fails on `main` CI baseline

## Summary

`tests/tools/test_provenance_execution.py::test_version_matches_path_scoped_commit_when_no_tag` fails on `main` under the CI selector:

```
uv sync --frozen --extra test
uv run pytest -m "not real_data and not fortran_binary" -q
```

This is the only failing test in the suite (the other ~1900 pass), so it's blocking a green CI baseline on `main`.

## Failure

```
FAILED tests/tools/test_provenance_execution.py::test_version_matches_path_scoped_commit_when_no_tag

>       assert version == path_commit
E       AssertionError: assert '0.1.0' == ''
E
E         + 0.1.0
```

## What the test does

```python
def test_version_matches_path_scoped_commit_when_no_tag(monkeypatch):
    repo_root = _package_repo_root()
    if repo_root is None:
        pytest.skip("not running in a git checkout")
    # Pretend no tags exist at HEAD by intercepting the describe call.
    real_git = execution._git

    def fake_git(args, root):
        if args[:3] == ["describe", "--exact-match", "--tags"]:
            return None
        return real_git(args, root)

    monkeypatch.setattr(execution, "_git", fake_git)
    version = _resolve_version()
    # Should equal the most recent commit touching src/lysis/.
    lysis_path = repo_root / "src" / "lysis"
    path_commit = subprocess.check_output(
        ["git", "-C", str(repo_root), "log", "-1", "--format=%H", "HEAD",
         "--", str(lysis_path)],
        text=True,
    ).strip()
    assert version == path_commit
```

The test monkey-patches `git describe --exact-match --tags` to return `None` (pretend HEAD has no tag), then expects `_resolve_version()` to fall back to "most recent commit touching `src/lysis/`".

## Diagnosis

Two things are happening:

1. **The expected side fails: `path_commit` is the empty string.** `git log -1 --format=%H HEAD -- src/lysis` is producing no output. That means in this checkout, there is no commit on `HEAD`'s history that touched `src/lysis/`. This is unusual but plausible if the test is running against a sparse / shallow checkout, a worktree where the path didn't exist when the recent commits were made, or — more likely on `main` right now — if the package source actually lives at a path other than `src/lysis/` for some commits in the history. (E.g. the legacy code was at `src/python/lysis/`, and commits before the move don't touch `src/lysis/`. With a shallow CI clone, the log can come up empty.)
2. **The actual side resolves to `'0.1.0'`.** That's the `version = "0.1.0"` from `pyproject.toml`, returned by the `importlib.metadata` fallback — meaning `_resolve_version()` already took the package-metadata path before reaching the path-scoped-commit branch.

So both halves of the equality are wrong relative to the test's intent: the fixture conditions for the "no tag, has commits touching src/lysis/" branch aren't actually met in the CI environment, and the resolver picks a different fallback.

## Likely fixes (pick one)

1. **Loosen the test:** if `path_commit` is empty (no qualifying commit in this checkout's history), `pytest.skip("no src/lysis/ history reachable in this checkout")`. Combine with the existing `_package_repo_root() is None` skip. This is the safest minimal change and matches how the test already handles "not a git checkout".
2. **Fix the resolver order:** if `_resolve_version` is supposed to prefer path-scoped-commit over `importlib.metadata` when a git checkout is present, audit `src/lysis/tools/provenance/execution.py` and the `_resolve_version` ordering. The test's monkeypatch only intercepts `describe`; if `_resolve_version` is short-circuiting to `importlib.metadata` before consulting `git log -- src/lysis`, that's a real bug to fix in the resolver, not the test.
3. **Use a deeper clone in CI:** add `fetch-depth: 0` to the `actions/checkout@v4` step in `.github/workflows/tests.yml` so the full history (and the path-scoped log) is available. This won't help if the test is also failing locally on a full clone — needs to be reproduced both ways.

Recommendation: reproduce locally on a full clone first (`git clone <fresh>` of `main`, then `uv sync --frozen --extra test && uv run pytest tests/tools/test_provenance_execution.py::test_version_matches_path_scoped_commit_when_no_tag -q`).
- If it **passes** on a full local clone, the fix is `fetch-depth: 0` in the workflow plus option (1) defensively.
- If it **fails** on a full local clone, the resolver is short-circuiting to `importlib.metadata` before the path-scoped lookup — fix in `execution._resolve_version`.

## Acceptance criteria

- [ ] `uv run pytest -m "not real_data and not fortran_binary" -q` is fully green on `main`.
- [ ] The fix doesn't reintroduce the "test silently passes by skipping everything" failure mode — if the test skips, it must skip with a specific, documented reason.
- [ ] If the workflow needed `fetch-depth: 0`, that change is documented in a comment alongside the step.
- [ ] If the resolver order changed, the docstring on `_resolve_version` is updated to match.

## Related

- CI workflow: `.github/workflows/tests.yml`
- Resolver under test: `src/lysis/tools/provenance/execution.py` (`_resolve_version`, `_package_repo_root`, `_git`)

--- a/tests/tools/test_provenance_execution.py
+++ b/tests/tools/test_provenance_execution.py
@@ -13,7 +13,6 @@ the once-per-process dirty-warning machinery, and the path-scoped
 """
 
 import re
-import subprocess
 import warnings
 
 import pytest
@@ -25,7 +24,6 @@ from lysis.tools.provenance import (
     gather_init_provenance,
     mark_dirty_warning_emitted,
 )
-from lysis.tools.provenance._git import _package_repo_root
 from lysis.tools.provenance.execution import (
     _resolve_dirty,
     _resolve_hostname,
@@ -71,28 +69,25 @@ def test_version_is_nonempty_string():
     assert _resolve_version()
 
 
-def test_version_matches_path_scoped_commit_when_no_tag(monkeypatch):
-    repo_root = _package_repo_root()
-    if repo_root is None:
-        pytest.skip("not running in a git checkout")
-    # Pretend no tags exist at HEAD by intercepting the describe call.
-    real_git = execution._git
+def test_version_matches_path_scoped_commit_when_no_tag(monkeypatch, tmp_path):
+    repo_root = tmp_path
+    lysis_path = repo_root / "src" / "lysis"
+    path_commit = "a" * 40
+
+    monkeypatch.setattr(execution, "_package_repo_root", lambda: repo_root)
 
     def fake_git(args, root):
-        if args[:3] == ["describe", "--exact-match", "--tags"]:
+        assert root == repo_root
+        if args == ["log", "-1", "--format=%H", "HEAD", "--", str(lysis_path)]:
+            return path_commit
+        if args == ["rev-parse", "HEAD"]:
+            return path_commit
+        if args == ["describe", "--exact-match", "--tags", "HEAD"]:
             return None
-        return real_git(args, root)
+        raise AssertionError(f"unexpected git call: {args}")
 
     monkeypatch.setattr(execution, "_git", fake_git)
-    version = _resolve_version()
-    # Should equal the most recent commit touching src/lysis/.
-    lysis_path = repo_root / "src" / "lysis"
-    path_commit = subprocess.check_output(
-        ["git", "-C", str(repo_root), "log", "-1", "--format=%H", "HEAD",
-         "--", str(lysis_path)],
-        text=True,
-    ).strip()
-    assert version == path_commit
+    assert _resolve_version() == path_commit
 
 
 def test_version_falls_back_to_package_metadata(monkeypatch):

